### DISCUSSION TITLE
Running "cases" to test installation

### DISCUSSION THREAD

**Discussion by ghost** [NOAA-EMC/WW3#564] [Q&A]
I just downloaded and installed WW3 for the first time (WAVEWATCH-III.v6.07.1) and I'm going through some of the "regtests" and "cases" to check the installation and get a feel for how to work with the model. I didn't run all the "regtests", but the ones I did run, seemed to work without hassle. The "cases" however are not working. For starters the "${ww3_dir}" needs to be adjusted. It seems the directory structure has since been updated by including "model" as the main parent directory for the source code.

I've been running test case #3 and I shut off MPI and reduced the "mods" to "'glo_30m'" only. I'm also only using the wind input. This was as simple as I could make the model to try an narrow down the problem, but I'm still not having any luck. I can run all the preprocessors without trouble, but when I try to run the main program ("ww3_multi") I get the following error

 *** WAVEWATCH III ERROR IN WMINIT : *** 
     ERROR IN READING FROM INPUT FILE
     IOSTAT =    0

I've systematically went through the "mww3_case_03" test script (see below) and ran each preprocessor w/o removing files. When I look at the temp working directory, I can see (in image below) the following list of files:
![image](https://user-images.githubusercontent.com/8746187/147397830-1910b429-8f9f-4dcf-b3c7-1b14e2de28a5.png)

I don't feel like a file is missing. There's nothing in the log file. I'm on a Linux machine, so I don't think there's any "dos2unix" issues. I've tested both serial and MPI executables, but I get the same error. Since the "regtests" work, I don't think it's a compiler issue (gcc 9.3).

Can someone help by suggesting methods to identify this error?

I've copy pasted the test scrip file below (had to make it an attachment due to text formatting here).
NOTE: In about a quarter of the way down, I redefine "${ww3_dir}" in which you'll need to insert your "PARENT_DIRECTORY" (removed mine)

[code.txt](https://github.com/NOAA-EMC/WW3/files/7777544/code.txt)

**Comment by kebrunner** [comment]
I'm also a new WW3 user, have been doing the same tests as you, and ran into the same exact problem with "regtest" case #3. I've been trying to troubleshoot this on my own for a couple months now and have not been able to find any one else that seems to have experienced this problem so it feels good to know that I'm not alone! I posted about this problem on NOAA's WW3 Users listserv (www.lstsrv.ncep.noaa.gov) at the end of January and got no response, so no help there. I'm going to continue to work on this, but hopefully a more experienced user might be able to identify what the problem with the regtests are.

**Reply by JessicaMeixner-NOAA** [reply]
Can you either post your email content here or re-send your email to the users group.  I can't seem to find it.

**Reply by kebrunner** [reply]
I've copied what I posted below:


Hi all,

I’m a new WW3 user working my way through some example cases before trying to run something of my own. I’ve successfully been able to run all of the provided test cases I’ve tried in the “regtests” directory. I’m now trying to run one of the more real-world test cases provided in the “cases” directory, namely mww3_case_03 which I would like to eventually modify for my own use, but have not been able to get it to successfully run. I’ve changed the compiler string for my system, as well as the case directory and WW3 directory paths. I’ve also unzipped the compressed data files. It runs fine until it gets to the ww3_multi step. I’ve copied the relevant error output to the screen below:

 Running multi-grid model ...

                    *** WAVEWATCH III Multi-grid shell ***
               =================================================

  Opening input file ww3_multi.inp (unit number  8)

  Opening output file log.mww3 (unit number  9)

  Comment character : '$'

  Number of grids          :  8

  Input data grids         :  2

  Single point output file : YES/--

  Output server type       :  1

  Single point output proc : ---/NO

 *** WAVEWATCH III ERROR IN WMINIT : ***

     ERROR IN READING FROM INPUT FILE

     IOSTAT =    0

   Log files routed to /WW3/cases/work_case_03


There appears to be some sort of input error, but there is no additional information provided (the log file is empty). Has anyone been able to successfully run this real-world test case or seen this particular error before and have any tips? Any information would be very helpful.

In an attempt to troubleshoot the issue, I’ve commented out all of the statements where the *.inp files and temporary directory get removed (for my education to get a better idea of how these files are formatted and see if there’s any noticeable error in there).  Thanks so much for your help!

**Reply by kebrunner** [reply]
@JessicaMeixner-NOAA I was wondering if you could maybe help me find or modify the right test case to help me learn what I'm trying to do. I have a lot of ocean modeling experience but for some reason Wavewatch just isn't clicking with me so I've been struggling to figure it out. I'm looking for a large domain with one or more subdomains that are set up using the multi-grid model driver input file. I'd also like to get wave spectra output at point locations. Reading the manual, in theory I understand how I should be able to modify one of the test cases, but it seems like the run_test script actually builds the input files and runs the test cases so I'm a little confused.

**Reply by JessicaMeixner-NOAA** [reply]
One of the regression tests could likely be used as an example.   The run_test script does actually build and run the test case but it can be used as an example to run things with out run_test as well.    

It's really difficult to port the cases to a machine without access, etc.  I believe @aliabdolali ran the cases before the last public release so perhaps he can provide some insight when he's back from leave next week.

**Reply by JessicaMeixner-NOAA** [reply]
@kebrunner Try mww3_test_03 with grid sets b d or e to see if that is a good example.

**Comment by JessicaMeixner-NOAA** [comment]
@[XxgithubxX](https://github.com/XxgithubxX) If the regression tests are working, then you should feel good that the model is installed correctly.   @kebrunner -- for a quick solution I'd recommend using the regression tests as it seems there's perhaps an issue or a machine dependent issue in the cases.    In general the cases are not run frequently and not tested as extensively as the regression tests.    I'll see if I can go back and find your users list email to see if I can get more specifics.

**Reply by kebrunner** [reply]
The regression tests are working fine, just the cases aren't working. I was hoping to get the cases working so that I could modify them as they are much closer to what I actually need to do with the model.

**Reply by ghost** [reply]
I agree with @kebrunner. The regression tests are great for development and cross checking installation, but they are not practical in an operational sense. The "cases" provide a template for operational use. This is exactly what most end users are interested in. It is therefore a shame that the "cases" don't work.  WW3 is already difficult enough from a user perspective, so not having a minimum working example for operation use is a drag...(I gave up 3 months ago)

**Reply by JessicaMeixner-NOAA** [reply]
Thanks for your feedback.  We are always doing our best to make WW3 more user friendly.   

I'd recommend trying mww3_test_03 test case for a multi-grid approach.  Hopefully @aliabdolali will have some hints on the cases when he returns from leave next week.
